p2panda-rs/src/storage_provider/traits/responses.rs

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use async_trait::async_trait;
+
+use crate::entry::{LogId, SeqNum};
+use crate::hash::Hash;
+
+/// Response when requesting arguments to create a new entry.
+#[async_trait]
+pub trait AsEntryArgsResponse {
+    /// Just the constructor method is defined here as all we need this trait for is constructing
+    /// entry arguments to be returned from the default trait methods.
+    ///
+    /// NB: We could validate that entries at seq nums that require skiplinks have them.
```

Comment:
Does this save us from validating that in some other place?

Reply:
Hmmm, not at the moment, as we do this ourselves in the default entry args method which is the only place a struct with this trait is constructed..... I suppose it would be to protect against implementors making mistakes if they use different patterns when constructing their entry arg responses.

Maybe this can be looked at in "The Big Validation Review".

Reply:
I'm adding this to the review issue :+1: